source/simulator/helper_functions.cc

```diff
@@ -1694,72 +1686,82 @@ namespace aspect
     // So even though we touch some DoF more than once, we always start from the same value, compute the
     // same value, and then overwrite the same value in distributed_vector.
     // TODO: make this more efficient.
+
+    using VectorType = Vector<double>;
+    SUNDIALS::ARKode<VectorType>::AdditionalData data (0.0, time_step, 0.001*time_step, 0.1*time_step, 1.e-6*time_step);
```

Comment:
This is hard to read. It corresponds to this function:
```
      AdditionalData(
        // Initial parameters
        const double initial_time      = 0.0,
        const double final_time        = 1.0,
        const double initial_step_size = 1e-2,
        const double output_period     = 1e-1,
        // Running parameters
        const double       minimum_step_size                     = 1e-6,
        const unsigned int maximum_order                         = 5,
        const unsigned int maximum_non_linear_iterations         = 10,
        const bool         implicit_function_is_linear           = false,
        const bool         implicit_function_is_time_independent = false,
        const bool         mass_is_time_independent              = false,
        const int          anderson_acceleration_subspace        = 3,
        // Error parameters
        const double absolute_tolerance = 1e-6,
        const double relative_tolerance = 1e-5);
```
Perhaps you can write this as
```suggestion
    SUNDIALS::ARKode<VectorType>::AdditionalData data (/* initial_time = */0.0, 
                                                                                                        /* end_time = */ time_step,
                                                                                                        etc etc etc);
```
?

Reply:
Separately, though, it would avoid potential bugs if you replaced the initial time zero with the *actual* current time, and the end time as current time + time step.

I would probably choose a larger initial time step than 0.001*time_step. Do you actually need any output every 0.1*time step? This introduces a time point that ARKode needs to hit, and if you don't need the actual output, just set that parameter to zero or equal to time_step.

Reply:
For the formatting: I had to place the comment at the end for the indentation to look reasonable.